src/client/components/SignUpComponent/index.js

```diff
@@ -0,0 +1,116 @@
+      'errorPassword': false,
+    };
+  }
+  submitHandler(ev) {
+    this.state = {
+      'status': 'being',
+      'error': '',
+      'errorUsername': false,
+      'errorEmail': false,
+      'errorPhone': false,
+      'errorFullname': false,
+      'errorPassword': false,
+    };
+    ev.preventDefault();
+    let userName = ev.target.elements.namedItem('username').value;
+    let email = ev.target.elements.namedItem('email').value;
+    let phone = ev.target.elements.namedItem('phone').value;
+    let fullName = ev.target.elements.namedItem('fullname').value;
+    let password = ev.target.elements.namedItem('password').value;
+    let obj = {
+      'username': userName,
+      'email': email,
+      'phone number': phone,
+      'full name': fullName,
+      'password': password,
+    };
+    let jsonData = JSON.stringify(obj);
+    let xhr = new XMLHttpRequest();
+    xhr.addEventListener('readystatechange', function() {
+      if (xhr.readyState === XMLHttpRequest.DONE) {
+        if (xhr.status === 400) {
+          let errorText = JSON.parse(xhr.responseText).error;
+          if (errorText === 'the username is void') {
+            this.setState({'errorUsername': true});
+          }
+          if (errorText === 'the email is void') {
+            this.setState({'errorEmail': true});
+          }
+          if (errorText === 'the email format is invalid') {
+            this.setState({'errorEmail': true});
+          }
+          if (errorText === 'the phone number is void') {
+            this.setState({'errorPhone': true});
+          }
+          if (errorText === 'the phone number is invalid') {
+            this.setState({'errorPhone': true});
+          }
+          if (errorText === 'the full name is void') {
+            this.setState({'errorFullname': true});
+          }
+          if (errorText === 'the full name is invalid') {
+            this.setState({'errorFullname': true});
+          }
+          if (errorText === 'the password is void') {
+            this.setState({'errorPassword': true});
+          }
+          if (errorText === 'the password is too short') {
+            this.setState({'errorPassword': true});
+          }
+          this.setState({'error': errorText});
+        }
+        if (xhr.status === 409) {
+          let errorText = JSON.parse(xhr.responseText).error;
+          if (errorText === 'username conflicts!') {
+            this.setState({'errorUsername': true});
+          }
+          if (errorText === 'email conflicts!') {
+            this.setState({'errorEmail': true});
+          }
+          if (errorText === 'phone number conflicts!') {
+            this.setState({'errorPhone': true});
+          }
+          this.setState({'error': errorText});
+        }
+        if (xhr.status === 500) {
+          let errorText = JSON.parse(xhr.responseText).error;
+          this.setState({'error': errorText});
+        }
+      }
+      this.setState({'status': 'being'});
```

Comment:
this state change is in wrong place, you change the state right after the first readystatechange event, but you just need to change when the state is DONE